frontend/src/component/project/Project/ProjectOverview.tsx

```diff
@@ -34,22 +35,62 @@ const StyledContentContainer = styled(Box)(() => ({
     minWidth: 0,
 }));
 
+const InfiniteProjectOverview = () => {
+    const projectId = useRequiredPathParam('projectId');
+    const { project, loading: projectLoading } = useProject(projectId, {
+        refreshInterval,
+    });
+    const [nextCursor, setNextCursor] = useState('');
+    const { features: searchFeatures, refetch, loading } = useFeatureSearch(nextCursor, projectId , { refreshInterval});
+    const { members, features, health, description, environments, stats } =
+        project;
+
+    return (
+        <StyledContainer>
+            <ProjectInfo
+                id={projectId}
+                description={description}
+                memberCount={members}
+                health={health}
+                features={features}
+                stats={stats}
+            />
+            <StyledContentContainer>
+                <ProjectStats stats={project.stats} />
+                <StyledProjectToggles>
+                    <ProjectFeatureToggles
+                        key={loading ? 'loading' : 'ready'}
+                        features={searchFeatures.features}
+                        environments={environments}
+                        loading={loading}
+                        onChange={refetch}
+                    />
+                </StyledProjectToggles>
+            </StyledContentContainer>
+        </StyledContainer>
+    );
+
+};
+
 const ProjectOverview = () => {
     const projectId = useRequiredPathParam('projectId');
     const projectName = useProjectNameOrId(projectId);
-    const { project, loading } = useProject(projectId, {
+    const { project, loading, refetch } = useProject(projectId, {
         refreshInterval,
     });
     const { members, features, health, description, environments, stats } =
         project;
     usePageTitle(`Project overview – ${projectName}`);
     const { setLastViewed } = useLastViewedProject();
     const featureSwitchRefactor = useUiFlag('featureSwitchRefactor');
+    const featureSearchFrontend = useUiFlag('featureSearchFrontend');
 
     useEffect(() => {
         setLastViewed(projectId);
     }, [projectId, setLastViewed]);
 
+    if(featureSearchFrontend) return <InfiniteProjectOverview />;
```

Comment:
This is a technique I decided to use to early exit with new code when the flag is on. Otherwise the old code continues.